Error building

I am using the latest compiler that was downloaded using the build script.

```
__KERNEL__ -o fmemory.tmp fmemory.S
ia16-elf-as -mtune=i8086  --32-segelf -o fmemory.o fmemory.tmp
rm fmemory.tmp
gcc -E -traditional -I/home/toncho/elks/include -I/home/toncho/elks/elks/include -I/home/toncho/elks/libc/include -DUTS_RELEASE=\"0.4.0\" -D__KERNEL__ -o peekpoke.tmp peekpoke.S
ia16-elf-as -mtune=i8086  --32-segelf -o peekpoke.o peekpoke.tmp
rm peekpoke.tmp
gcc -E -traditional -I/home/toncho/elks/include -I/home/toncho/elks/elks/include -I/home/toncho/elks/libc/include -DUTS_RELEASE=\"0.4.0\" -D__KERNEL__ -o string.tmp string.S
ia16-elf-as -mtune=i8086  --32-segelf -o string.o string.tmp
rm string.tmp
gcc -E -traditional -I/home/toncho/elks/include -I/home/toncho/elks/elks/include -I/home/toncho/elks/libc/include -DUTS_RELEASE=\"0.4.0\" -D__KERNEL__ -o bios15.tmp bios15.S
ia16-elf-as -mtune=i8086  --32-segelf -o bios15.o bios15.tmp
rm bios15.tmp
gcc -E -traditional -I/home/toncho/elks/include -I/home/toncho/elks/elks/include -I/home/toncho/elks/libc/include -DUTS_RELEASE=\"0.4.0\" -D__KERNEL__ -o unreal.tmp unreal.S
ia16-elf-as -mtune=i8086  --32-segelf -o unreal.o unreal.tmp
unreal.S: Assembler messages:
unreal.S:32: Error: no such architecture: `'
unreal.S:32: Error: junk at end of line, first unrecognized character is `1'
unreal.S:341: Warning: stand-alone `addr32' prefix
unreal.S:391: Warning: stand-alone `addr32' prefix
unreal.S:398: Warning: stand-alone `addr32' prefix
../../../Makefile-rules:235: recipe for target 'unreal.o' failed
make[3]: *** [unreal.o] Error 1
make[3]: Leaving directory '/home/toncho/elks/elks/arch/i86/lib'
Makefile:191: recipe for target 'lib/lib86.a' failed
make[2]: *** [lib/lib86.a] Error 2
make[2]: Leaving directory '/home/toncho/elks/elks/arch/i86'
Makefile:73: recipe for target 'Image' failed
make[1]: *** [Image] Error 2
make[1]: Leaving directory '/home/toncho/elks/elks'
Makefile:11: recipe for target 'all' failed
make: *** [all] Error 2
Build script has terminated with error 5
```

--- a/elks/Makefile-rules
+++ b/elks/Makefile-rules
@@ -232,7 +232,7 @@ endif
 .S.s: ;
 
 .S.o:
-	gcc -E -traditional $(INCLUDES) $(CCDEFS) -o $*.tmp $<
+	$(CC) -E -traditional $(INCLUDES) $(CCDEFS) -o $*.tmp $<
 	$(AS) $(ASFLAGS) -o $*.o $*.tmp
 	rm $*.tmp
 

--- a/elks/arch/i86/lib/unreal.S
+++ b/elks/arch/i86/lib/unreal.S
@@ -294,8 +294,7 @@ linear32_peekw:
 	mov    %bx,%ds
 
 	cld
-	addr32                // 32-bit address prefix
-	lodsw                 // AX = [DS:ESI++]
+	addr32 lodsw          // AX = [DS:ESI++]
 	//popf                // restore interrupt status
 
 	mov    %dx,%si
@@ -338,11 +337,8 @@ linear32_fmemcpyw:
 	mov    %bx,%ds
 
 	cld
-	addr32                // 32-bit address prefix
-	rep
-	movsw                 // word [ES:EDI++] <- [DS:ESI++], ECX times
-	.byte  0x67           // 80386 B1 step chip bug on address size mixing
-	nop
+	addr32 rep movsw      // word [ES:EDI++] <- [DS:ESI++], ECX times
+	addr32 nop            // 80386 B1 step chip bug on address size mixing
 
 	//popf                // restore interrupt status
 
@@ -388,18 +384,12 @@ linear32_fmemcpyb:
 
 	cld
 	shrl   $1,%ecx        // copy words
-	addr32                // 32-bit address prefix
-	rep
-	movsw                 // word [ES:EDI++] <- [DS:ESI++], ECX times
-	.byte  0x67           // 80386 B1 step chip bug on address size mixing
-	nop
+	addr32 rep movsw      // word [ES:EDI++] <- [DS:ESI++], ECX times
+	addr32 nop            // 80386 B1 step chip bug on address size mixing
 
 	rcll    $1,%ecx       // then possibly final byte
-	addr32                // 32-bit address prefix
-	rep
-	movsb                 // byte [ES:EDI++] <- [DS:ESI++], ECX times
-	.byte  0x67           // 80386 B1 step chip bug on address size mixing
-	nop
+	addr32 rep movsb      // byte [ES:EDI++] <- [DS:ESI++], ECX times
+	addr32 nop            // 80386 B1 step chip bug on address size mixing
 
 	//popf                // restore interrupt status
 

--- a/elkscmd/Make.defs
+++ b/elkscmd/Make.defs
@@ -98,7 +98,7 @@ CFLAGS=$(CFLBASE) $(WARNINGS) $(LOCALFLAGS) $(INCLUDES) -D__ELKS__ -DELKS_VERSIO
 # Standard compilation rules.
 
 .S.s:
-	gcc -E -traditional $(INCLUDES) $(CCDEFS) -o $*.s $<
+	$(CC) -E -traditional $(INCLUDES) $(CCDEFS) -o $*.s $<
 
 .s.o:
 	$(AS) $(ASFLAGS) -o $*.o $<

